### ISSUE TITLE
TestContainers for Local Dev

### ISSUE THREAD

**Issue by jamesward** [cloudstateio/cloudstate#369]
When doing local development the proxy can be started with the following to connect to the host's function port:
```
docker run -it --rm -p9000:9000 --network="host" cloudstateio/cloudstate-proxy-dev-mode -Dcloudstate.proxy.user-function-port=8080
```

Which is cool and should be documented somewhere.  But also this means that TestContainers could be used for local dev / testing, which would be extra cool.

**Comment by sleipnir** [comment]
Hello @jamesward In fact it is very interesting and an approach that has already been used in support languages like here for example:

https://github.com/cloudstateio/kotlin-support/blob/master/integration-test/src/test/kotlin/io/cloudstate/kotlinsupport/tests/IntegrationTest.kt

**Comment by marcellanz** [comment]
@jamesward yes this is pretty cool and little known documentation wise. I also use this image quite a [lot for language support development](https://github.com/cloudstateio/go-support/tree/feature/crdt_support/tck) running the proxy and also the TCK and it's very helpful.
